source/adapters/level_zero/event.cpp

```diff
@@ -641,9 +641,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventWait(
       // This ensures that all signalling commands are submitted below and
       // thus proxy events can be waited without a deadlock.
       //
-      ur_event_handle_t_ *Event = ur_cast<ur_event_handle_t_ *>(e);
-      if (!Event->hasExternalRefs())
-        die("urEventsWait must not be called for an internal event");
+      ur_event_handle_t_ *Event =
```

Comment:
why drop the use of the variable e? then it will just be unused in the internals.